src/foam/build/FilesJsGen.js

```diff
@@ -0,0 +1,290 @@
+    },
+  ],
+  methods: [
+    function execute() {
+      this.getFilesJs().then(console.log);
+    },
+    function getFilesJs() {
+      var self = this;
+      var getTreeHead = function(pred) {
+        return self.modelDAO
+          .where(pred)
+          .select()
+          .then(function(s) {
+            return self.getDepsTree(s.array)
+          });
+      };
+      return Promise.all([
+          getTreeHead(self.IN(self.Model.ID, self.CORE_MODELS)),
+          getTreeHead(self.INSTANCE_OF(self.Lib)),
+          getTreeHead(self.IN(self.Model.ID, self.PHASE_1)),
+          getTreeHead(self.IN(self.Model.ID, self.PHASE_2)),
+          getTreeHead(self.INSTANCE_OF(self.Script)),
+          getTreeHead(self.INSTANCE_OF(self.Relationship)),
+          getTreeHead(self.HAS(self.Model.REFINES)),
+          getTreeHead(self.IN(self.Model.ID, self.required)),
+      ]).then(function(args) {
+        return Promise.all(
+          args.map(function(head) {
+            var order = [];
+            var queue = [head];
+            while ( queue.length ) {
+              var n = queue.pop();
+              Object.keys(n).forEach(function(k) {
+                if ( order.indexOf(k) == -1 ) queue.unshift(n[k]);
+                order.unshift(k)
+              });
+            }
+            // Remove anyting not in the DAO. This can happen for inner classes
+            // and enums. TODO: Would be nice if we didn't have to do this.
+            return Promise.all(order.map(function(id) {
+              return self.modelDAO.find(id);
+            })).then(function(models) {
+              return models.filter(function(m) { return !!m }).map(function(m) { return m.id });
+            });
+          })
+        );
+      }).then(function(args) {
+        // Flatten args.
+        args = [].concat.apply([], args);
+        var files = [].concat(
+          self.BOOT_FILES,
+          self.CORE_MODELS,
+          args).map(function(o) {
+            return `{ name: "${o.replace(/\./g, '/')}" },`;
+          });
+
+        // Remove duplicates.
+        files = files.filter(function(id, i) {
+          return files.indexOf(id) == i;
+        })
+
+        var filesJs = `
+if ( typeof window !== 'undefined' ) global = window;
+FOAM_FILES([
+  ${files.join('\n  ')}
+]);
+        `.trim();
+        return filesJs
+      });
+    },
+    function getDepsTree(o, seen, head) {
+      var self = this;
+      seen = seen || {};
+      head = head || {};
+
+      if ( foam.Array.isInstance(o) ) {
+        return Promise.all(o.map(function(i) {
+          return self.getDepsTree(i, seen, head);
+        })).then(function() { return head });
+      } else if ( foam.Object.isInstance(o) ) {
+        // Check if it's an actual class. foam.core.FObject.isSubClass
+        // should work but doesn't:
+        // https://github.com/foam-framework/foam2/issues/1023
+        if ( o && o.prototype && (
+            foam.core.FObject.prototype === o.prototype ||
```

Comment:
Indentation here is confusing, move the open parenthesis to this line so it read

```
if ( o && o.prototype &&
     ( foam.core.FObject... ||
       foam.core.FObject... ) ) {
```

Reply:
Ah yeah that's nicer. Fixed here and in the JsCodeOutputter.